internal/services/paloalto/validate/country_code.go

```diff
@@ -0,0 +1,274 @@
+package validate
+
+import "fmt"
+
+var ISO3166_1_alpha2 = []string{
+	"AF",
+	"AX",
+	"AL",
+	"DZ",
+	"AS",
+	"AD",
+	"AO",
+	"AI",
+	"AQ",
+	"AG",
+	"AR",
+	"AM",
+	"AW",
+	"AU",
+	"AT",
+	"AZ",
+	"BS",
+	"BH",
+	"BD",
+	"BB",
+	"BY",
+	"BE",
+	"BZ",
+	"BJ",
+	"BM",
+	"BT",
+	"BO",
+	"BQ",
+	"BA",
+	"BW",
+	"BV",
+	"BR",
+	"IO",
+	"BN",
+	"BG",
+	"BF",
+	"BI",
+	"KH",
+	"CM",
+	"CA",
+	"CV",
+	"KY",
+	"CF",
+	"TD",
+	"CL",
+	"CN",
+	"CX",
+	"CC",
+	"CO",
+	"KM",
+	"CG",
+	"CD",
+	"CK",
+	"CR",
+	"CI",
+	"HR",
+	"CU",
+	"CW",
+	"CY",
+	"CZ",
+	"DK",
+	"DJ",
+	"DM",
+	"DO",
+	"EC",
+	"EG",
+	"SV",
+	"GQ",
+	"ER",
+	"EE",
+	"ET",
+	"FK",
+	"FO",
+	"FJ",
+	"FI",
+	"FR",
+	"GF",
+	"PF",
+	"TF",
+	"GA",
+	"GM",
+	"GE",
+	"DE",
+	"GH",
+	"GI",
+	"GR",
+	"GL",
+	"GD",
+	"GP",
+	"GU",
+	"GT",
+	"GG",
+	"GN",
+	"GW",
+	"GY",
+	"HT",
+	"HM",
+	"VA",
+	"HN",
+	"HK",
+	"HU",
+	"IS",
+	"IN",
+	"ID",
+	"IR",
+	"IQ",
+	"IE",
+	"IM",
+	"IL",
+	"IT",
+	"JM",
+	"JP",
+	"JE",
+	"JO",
+	"KZ",
+	"KE",
+	"KI",
+	"KP",
+	"KR",
+	"KW",
+	"KG",
+	"LA",
+	"LV",
+	"LB",
+	"LS",
+	"LR",
+	"LY",
+	"LI",
+	"LT",
+	"LU",
+	"MO",
+	"MK",
+	"MG",
+	"MW",
+	"MY",
+	"MV",
+	"ML",
+	"MT",
+	"MH",
+	"MQ",
+	"MR",
+	"MU",
+	"YT",
+	"MX",
+	"FM",
+	"MD",
+	"MC",
+	"MN",
+	"ME",
+	"MS",
+	"MA",
+	"MZ",
+	"MM",
+	"NA",
+	"NR",
+	"NP",
+	"NL",
+	"NC",
+	"NZ",
+	"NI",
+	"NE",
+	"NG",
+	"NU",
+	"NF",
+	"MP",
+	"NO",
+	"OM",
+	"PK",
+	"PW",
+	"PS",
+	"PA",
+	"PG",
+	"PY",
+	"PE",
+	"PH",
+	"PN",
+	"PL",
+	"PT",
+	"PR",
+	"QA",
+	"RE",
+	"RO",
+	"RU",
+	"RW",
+	"BL",
+	"SH",
+	"KN",
+	"LC",
+	"MF",
+	"PM",
+	"VC",
+	"WS",
+	"SM",
+	"ST",
+	"SA",
+	"SN",
+	"RS",
+	"SC",
+	"SL",
+	"SG",
+	"SX",
+	"SK",
+	"SI",
+	"SB",
+	"SO",
+	"ZA",
+	"GS",
+	"SS",
+	"ES",
+	"LK",
+	"SD",
+	"SR",
+	"SJ",
+	"SZ",
+	"SE",
+	"CH",
+	"SY",
+	"TW",
+	"TJ",
+	"TZ",
+	"TH",
+	"TL",
+	"TG",
+	"TK",
+	"TO",
+	"TT",
+	"TN",
+	"TR",
+	"TM",
+	"TC",
+	"TV",
+	"UG",
+	"UA",
+	"AE",
+	"GB",
+	"US",
+	"UM",
+	"UY",
+	"UZ",
+	"VU",
+	"VE",
+	"VN",
+	"VG",
+	"VI",
+	"WF",
+	"EH",
+	"YE",
+	"ZM",
+	"ZW",
+}
+
+// ISO3361CountryCode checks that a supplied country code is pesent in the ISO3361-1 alpha 2 country code list as
+// defined at http://www.ip2country.net/ip2country/country_code.html
+func ISO3361CountryCode(input interface{}, k string) (warnings []string, errors []error) {
+	v, ok := input.(string)
+	if !ok {
+		errors = append(errors, fmt.Errorf("expected %s to be of type string", k))
+		return
+	}
+
+	for _, a := range ISO3166_1_alpha2 {
+		if a == v {
+			return
+		}
+	}
+	errors = append(errors, fmt.Errorf("%s (%q) is not a valid ISO3361-1 Alpha-2 country code, see the official Palo Alto supported list at https://knowledgebase.paloaltonetworks.com/KCSArticleDetail?id=kA10g000000ClFFCA0 for more information", k, v))
```

Comment:
that list may differ from what's in the provider fwiw, should we output this list in some form?

Reply:
This list was generated from the PA spec, I don't think we can be more accurate here?